File: examples/python/basic_usage.py
import os
from remembr import AgentMemoryClient

# Assuming you set REMEMBR_AGENT_TOKEN in your environment
token = os.environ.get("REMEMBR_AGENT_TOKEN")
if not token:
    raise ValueError("Please set REMEMBR_AGENT_TOKEN")

client = AgentMemoryClient(token=token)

# Store a memory
memory = client.store(
    value="The user prefers Python for scripting.",
    key="user_preference_lang",
    tags=["user", "preferences"],
    ttl="7d"
)
print("Stored memory:", memory.id)

# Search your memories
results = client.search(query="What is the user's preferred language?", limit=3)
print("\nSearch results:")
for r in results:
    print(f"- {r.value} (Score: {r.similarity})")

# List memories with tags
list_results = client.list(tags=["preferences"])
print("\nFiltered by tag:")
for r in list_results.data:
    print(f"- {r.key}: {r.value}")

File: examples/node/basic_usage.mjs
import { AgentMemoryClient } from '@remembr/sdk';

// Initialize the client using the REMEMBR_AGENT_TOKEN environment variable
const client = new AgentMemoryClient(process.env.REMEMBR_AGENT_TOKEN);

async function main() {
    // Store a memory
    const memory = await client.store({
        value: "The user prefers TypeScript for frontend work.",
        key: "user_preference_ts",
        tags: ["frontend", "user-preference"],
        ttl: "30d"
    });
    console.log(`Stored memory ID: ${memory.id}`);

    // Retrieve a memory by key
    const fetched = await client.get('user_preference_ts');
    console.log(`Fetched memory: ${fetched.value}`);

    // Search your memories semantically
    const results = await client.search("What language does the user like for UI?", { limit: 2 });
    console.log("Search results:");
    for (const r of results) {
        console.log(`- ${r.value} (Score: ${r.similarity})`);
    }
}

main().catch(console.error);
